pkg/operator2/route.go

```diff
@@ -18,13 +18,43 @@ func (c *authOperator) handleRoute() (*routev1.Route, error) {
 	if err != nil {
 		return nil, err
 	}
-	if len(route.Spec.Host) == 0 {
-		return nil, fmt.Errorf("route has no host: %#v", route)
+
+	if err := isValidRoute(route); err != nil {
```

Comment:
Should we try to reconcile the route if it does not meet our expectations?

Reply:
You could just add a delete for the route.  The validation error will cause the loop to retried and the route will be fixed on the next iteration.